### PR TITLE
Renaming blacklist/whitelist aligned with iOS #7557

### PR DESCRIPTION
**Pull request by VMadalin**
### Description
https://github.com/mozilla-mobile/android-components/issues/7557

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### Overview
I follow exactly the iOS changes on this PR: https://github.com/mozilla-mobile/firefox-ios/pull/6783/files

### CODE REVIEW

**Review by Amejia481 — COMMENTED** [review]
Thanks for this patch!
It look pretty good, I just left some questions apart from that I looks good 👍🏽

File: components/feature/addons/src/test/resources/collection_with_empty_values.json [hunk 1]
```diff
@@ -72,7 +72,7 @@
...
...c1919e4\" rel=\"nofollow\">Fix icon title always showing <code>(0)</code> when badge is disabled</a></li><li><a href=\"https://outgoing.prod.mozaws.net/v1/8d3f56432a85938bde956203be79a070723f2555e49cb46b6d50a9dcbd53768d/https%3A//github.com/gorhill/uBlock/commit/1e7e6f86a6b57ad5e962e7196108e9d6a676bc04\" rel=\"nofollow\">Reuse existing Set/Map when calling scriptletFilteringEngine.retrieve</a></li><li><a href=\"https://outgoing.prod.mozaws.net/v1/4e26b4ec5cf051d2997115a58a6360af94abee5e4376fc4475bd2123b3c2cea6/https%3A//github.com/gorhill/uBlock/commit/a73dd0a9f26df59cfb4b68e98d1184ac042c90ea\" rel=\"nofollow\">Fix entity-based lookup in html &amp; scriptlet filtering</a></li><li><a href=\"https://outgoing.prod.mozaws.net/v1/181c7b80be2a8ffb41444d1d6063116d677b8eb39eb025a1ae499371e1adcfac/https%3A//github.com/gorhill/uBlock/commit/4792e0e291221c0f3aa158b759c9bbb16b788785\" rel=\"nofollow\">Coalesce tab reloads in burst \"relax blocking mode\" ops</a></li></ul>\n<a href=\"https://outgoing.prod.mozaws.net/v1/4c13192c5873c48eb61a007dc7a7cc495fdfb9a7ed57faa83e5bef457c478c4d/https%3A//github.com/gorhill/uBlock/compare/1.22.4...1.23.0\" rel=\"nofollow\">Commits history since 1.22.4</a>."
```
[the remark this answers is not in the document]
Comment (VMadalin) [reply components/feature/addons/src/test/resources/collection_with_empty_values.json:75]: Let me know if we don't want to apply changes on this file but the bellow is the change:
`Convert new setTimeout-if scriptlet to blacklist approach` to `Convert new setTimeout-if scriptlet to blocklist approach`

File: components/feature/addons/src/test/resources/collection.json [hunk 2]
```diff
@@ -89,7 +89,7 @@
...
...c1919e4\" rel=\"nofollow\">Fix icon title always showing <code>(0)</code> when badge is disabled</a></li><li><a href=\"https://outgoing.prod.mozaws.net/v1/8d3f56432a85938bde956203be79a070723f2555e49cb46b6d50a9dcbd53768d/https%3A//github.com/gorhill/uBlock/commit/1e7e6f86a6b57ad5e962e7196108e9d6a676bc04\" rel=\"nofollow\">Reuse existing Set/Map when calling scriptletFilteringEngine.retrieve</a></li><li><a href=\"https://outgoing.prod.mozaws.net/v1/4e26b4ec5cf051d2997115a58a6360af94abee5e4376fc4475bd2123b3c2cea6/https%3A//github.com/gorhill/uBlock/commit/a73dd0a9f26df59cfb4b68e98d1184ac042c90ea\" rel=\"nofollow\">Fix entity-based lookup in html &amp; scriptlet filtering</a></li><li><a href=\"https://outgoing.prod.mozaws.net/v1/181c7b80be2a8ffb41444d1d6063116d677b8eb39eb025a1ae499371e1adcfac/https%3A//github.com/gorhill/uBlock/commit/4792e0e291221c0f3aa158b759c9bbb16b788785\" rel=\"nofollow\">Coalesce tab reloads in burst \"relax blocking mode\" ops</a></li></ul>\n<a href=\"https://outgoing.prod.mozaws.net/v1/4c13192c5873c48eb61a007dc7a7cc495fdfb9a7ed57faa83e5bef457c478c4d/https%3A//github.com/gorhill/uBlock/compare/1.22.4...1.23.0\" rel=\"nofollow\">Commits history since 1.22.4</a>."
```
[the remark this answers is not in the document]
Comment (VMadalin) [reply components/feature/addons/src/test/resources/collection.json:92]: Let me know if we don't want to apply changes on this file but the bellow is the change:
`Convert new setTimeout-if scriptlet to blacklist approach` to `Convert new setTimeout-if scriptlet to blocklist approach`

File: components/feature/webcompat/src/main/assets/extensions/webcompat/experiment-apis/matchPatterns.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
         "parameters": [
           {
             "name": "patterns",
-            "description": "Array of string URL patterns to whitelist",
+            "description": "Array of string URL patterns to safelist",
```
[the remark this answers is not in the document]
Comment (denschub) [reply components/feature/webcompat/src/main/assets/extensions/webcompat/experiment-apis/matchPatterns.json:13]: Even better, https://github.com/mozilla-extensions/webcompat-addon is our "upstream" repo!

Good catch, thanks so much. Feel free to file a bug in our GitHub project, or if you want, you can also submit a PR! (although "allowlist" would probably be better here, this isn't a security-context in this case :D)

File: components/feature/addons/src/test/resources/collection.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Amejia481) [reply components/feature/addons/src/test/resources/collection.json:92]: It's not a problem. I was just curious to know, as this test file. It bit difficult to know what have changed as has a long  horizontal scroll.

File: components/feature/addons/src/test/resources/collection_with_empty_values.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Amejia481) [reply components/feature/addons/src/test/resources/collection_with_empty_values.json:75]: It's not a problem. I was just curious to know, as this test file. It bit difficult to know what have changed as has a long  horizontal scroll.

**Review by Amejia481 — COMMENTED** [review]
Thanks for the hard work @VMadalin, we can land this after we add the changelog entry and fix the merge conflicts 👍🏽

**Comment by Amejia481** [comment]
Awesome work @VMadalin could you squash the commits? Then we can 🚢 it! :)

File: components/feature/webcompat/src/main/assets/extensions/webcompat/experiment-apis/matchPatterns.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (VMadalin) [reply components/feature/webcompat/src/main/assets/extensions/webcompat/experiment-apis/matchPatterns.json:13]: Done https://github.com/mozilla-extensions/webcompat-addon/pull/195

**Review by Amejia481 — APPROVED** [review]

**Pull request merged**